### DISCUSSION TITLE
Deploying React SSR Application on AWS ec2 instance

### DISCUSSION THREAD

**Discussion by jagadeeshgade008** [vikejs/vike#997] [Discussion] [answered]
I have implemented my project using npm run dev.
How can I deploy and host this application in the **aws ec2** instance?

**Comment by brillout** [comment, marked as answer]
I've improved the AWS docs: https://vite-plugin-ssr.com/aws. Let me know if you still have questions.

**Reply by jagadeeshgade008** [reply]
I have built with **npm run build** dist folder created successfully!

![image](https://github.com/brillout/vite-plugin-ssr/assets/79077571/63e06051-fa95-4962-bd01-3659fa53a4ed)

getting the following error after running a build with **npm run server:prod** this command

![image](https://github.com/brillout/vite-plugin-ssr/assets/79077571/87dbcab0-b124-4a0e-9ecc-fa22351c0f6d)

**Reply by brillout** [reply]
Minimal reproduction welcome and I'll then have a look at it.

**Reply by jagadeeshgade008** [reply]
is this error some thing related to node modules or node version i am using?
i am using node v18.12.1

**Reply by brillout** [reply]
> Minimal reproduction

**Reply by brillout** [reply]
The best would be a minimal reproduction.

**Reply by jagadeeshgade008** [reply]
I initiated the project with **npm init vite-plugin-ssr** after that I installed the necessary node modules and made a production build with **npm run build** then i ran the application with **npm run server:prod** application ran properly but when I try to open page getting this error
TypeError: jsxDEV is not a function

**Reply by brillout** [reply]
That's quite unexpected. Let me try to reproduce.

**Reply by brillout** [reply]
I can't reproduce with `v18.15.0`. Can you try with that version?

**Reply by brillout** [reply]
Node `v18.15.0`.

**Reply by jagadeeshgade008** [reply]
adding NODE_ENV=production in build command resolved my issue
